post.js

```diff
@@ -39,6 +40,48 @@ function timeSince (dateString) {
   return Math.floor(seconds) + 's'
 }
 
+function insertImagesAndVideos (content) {
+  const parser = new DOMParser()
+  const contentDOM = parser.parseFromString(content, 'text/html')
+
+  contentDOM.querySelectorAll('img').forEach(img => {
+    const originalSrc = img.getAttribute('src')
+    const p2pSrc = resolveP2PUrl(originalSrc)
+
+    img.onerror = () => {
+      console.log(`Failed to load image at ${originalSrc}. Attempting to resolve...`)
+      // Fallback to gateway URL only if it's a P2P URL and not already using a gateway
+      if (isP2P(originalSrc) && !originalSrc.includes('hypha.coop')) {
+        const fallbackSrc = resolveP2PUrl(originalSrc)
+        console.log(`Resolving to gateway URL due to error: ${fallbackSrc}`)
+        img.setAttribute('src', fallbackSrc)
+      }
+    }
+
+    img.setAttribute('src', p2pSrc)
+    console.log(`Set image src to: ${p2pSrc}`)
+  })
+
+  contentDOM.querySelectorAll('video source').forEach(video => {
+    const originalSrc = video.getAttribute('src')
+    const p2pSrc = resolveP2PUrl(originalSrc)
+
+    video.onerror = () => {
+      console.log(`Failed to load video at ${originalSrc}. Attempting to resolve...`)
+      if (isP2P(originalSrc) && !originalSrc.includes('hypha.coop')) {
+        const fallbackSrc = resolveP2PUrl(originalSrc)
+        console.log(`Resolving to gateway URL due to error: ${fallbackSrc}`)
+        video.setAttribute('src', fallbackSrc)
+      }
+    }
+
+    video.setAttribute('src', p2pSrc)
+    console.log(`Set video src to: ${p2pSrc}`)
+  })
+
+  return contentDOM.body.innerHTML // Return the modified HTML to be inserted
```

Comment:
Do the onerror events actually fire here? I think we need to add these listeners *after* we've sanitized and rendered the html for the post.